src/store/index.js

```diff
@@ -90,61 +92,32 @@ export default new Vuex.Store({
       navigator.geolocation.getCurrentPosition(onSuccess, onError, options);
     },
 
-    getCurrentPositionKey({ commit, state, dispatch }) {
-      const currentPositionAPIUrl = utils.getCurrentPositionAPIUrl({
-        latitude: state.currentPosition.latitude,
-        longitude: state.currentPosition.longitude,
-        APIkey: process.env.VUE_APP_ACCUWEATHER_KEY,
-      });
-      fetch(currentPositionAPIUrl)
-        .then(response => {
-          if (response.ok) return response.json();
-          commit(mutationTypes.SAVE_ERROR_DESC, response.statusText);
-          throw new Error(`HTTP error, status = ${response.status}`);
-        })
-        .then(positionJson => {
-          if (positionJson.Key) {
-            commit(mutationTypes.SAVE_CURRENT_POSITION_DATA, {
-              Key: positionJson.Key,
-              City: positionJson.LocalizedName,
-              dataLoadedFromAPI: true,
-            });
-            dispatch('getCurrentWeatherData');
-            dispatch('getHourlyForecastForCurrentLocation');
-            return;
-          }
-          commit(mutationTypes.SAVE_ERROR_DESC, 'Помилка при отриманні поточної погоди');
-        });
+    async getCurrentPosition({ commit, state, dispatch }) {
+      const currentPositionData = await saveCurrentPosition(state);
```

Comment:
This function `saveCurrentPosition` should probably throw an error. Passing an error as part of the payload is considered to be problematic